packages/gatsby/cache-dir/production-app.js

```diff
@@ -70,15 +70,16 @@ apiRunnerAsync(`onClientEntry`).then(() => {
         <Location>
           {({ location }) => (
             <EnsureResources location={location}>
-              {({ pageResources, location }) => (
-                <StaticQueryContext.Provider
-                  value={pageResources.staticQueryResults}
-                >
-                  <DataContext.Provider value={{ pageResources, location }}>
-                    {children}
-                  </DataContext.Provider>
-                </StaticQueryContext.Provider>
-              )}
+              {({ pageResources, location }) => {
+                const staticQueryResults = publicLoader.getStaticQueryResults()
+                return (
+                  <StaticQueryContext.Provider value={staticQueryResults}>
```

Comment:
This keeps the `StaticQueryContext` public API and also should be a lot more stable (fixing cases like we saw for `gatsby-plugin-transition-link`)